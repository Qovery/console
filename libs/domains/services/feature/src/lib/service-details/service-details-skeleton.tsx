import { type ComponentPropsWithoutRef } from 'react'
import { DescriptionDetails as Dd, DescriptionListRoot as Dl, DescriptionTerm as Dt, Skeleton } from '@qovery/shared/ui'
import { twMerge } from '@qovery/shared/util-js'

interface ServiceDetailsSkeletonProps extends ComponentPropsWithoutRef<'div'> {}

export function ServiceDetailsSkeleton({ className, ...props }: ServiceDetailsSkeletonProps) {
  return (
    <div className={twMerge('flex shrink-0 flex-col content-start justify-start gap-5 py-8', className)} {...props}>
      <div className="flex flex-col gap-4 pl-8 pr-5">
        <Skeleton height={26} width={50} show />
        <Skeleton height={16} width={300} show />
        <Dl>
          <Dt>
            <Skeleton height={16} width={75} show />
          </Dt>
          <Dd>
            <Skeleton height={16} width={75} show />
          </Dd>
          <Dt>
            <Skeleton height={16} width={75} show />
          </Dt>
          <Dd>
            <Skeleton height={16} width={75} show />
          </Dd>
        </Dl>
      </div>
      <hr />
      <div className="flex flex-col gap-4 pl-8 pr-5">
        <Skeleton height={26} width={50} show />
        <Skeleton height={16} width={300} show />
        <Dl>
          <Dt>
            <Skeleton height={16} width={75} show />
          </Dt>
          <Dd>
            <Skeleton height={16} width={75} show />
          </Dd>
          <Dt>
            <Skeleton height={16} width={75} show />
          </Dt>
          <Dd>
            <Skeleton height={16} width={75} show />
          </Dd>
        </Dl>
      </div>
    </div>
  )
}

export default ServiceDetailsSkeleton
