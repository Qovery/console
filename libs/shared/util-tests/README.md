# shared-util-tests

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared-util-tests` to execute the unit tests via [Jest](https://jestjs.io).
