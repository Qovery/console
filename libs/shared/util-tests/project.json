{
  "name": "shared-util-tests",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared/util-tests/src",
  "projectType": "library",
  "tags": ["scope:shared", "type:util"],
  "implicitDependencies": ["!shared-ui"],
  "targets": {
    "test": {
      "options": {
        "passWithNoTests": true
      },
      "configurations": {
        "ci": {
          "ci": true,
          "coverage": true
        }
      }
    }
  }
}
