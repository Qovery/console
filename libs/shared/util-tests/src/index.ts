export * from './lib/shared-util-tests'
