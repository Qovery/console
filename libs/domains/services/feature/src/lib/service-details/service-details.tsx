import { type ApplicationGitRepository, type Credentials } from 'qovery-typescript-axios'
import { type ComponentPropsWithoutRef } from 'react'
import { useParams } from 'react-router-dom'
import { P, match } from 'ts-pattern'
import { type Application, type Helm, type Job } from '@qovery/domains/services/data-access'
import {
  IconEnum,
  ServiceTypeEnum,
  isHelmGitSource,
  isHelmRepositorySource,
  isJobContainerSource,
  isJobGitSource,
} from '@qovery/shared/enums'
import {
  APPLICATION_SETTINGS_GENERAL_URL,
  APPLICATION_SETTINGS_RESOURCES_URL,
  APPLICATION_SETTINGS_URL,
  APPLICATION_SETTINGS_VALUES_OVERRIDE_FILE_URL,
} from '@qovery/shared/routes'
import {
  Badge,
  Button,
  DescriptionDetails as Dd,
  DescriptionListRoot as Dl,
  DescriptionTerm as Dt,
  Icon,
  Link,
  Section,
  ToastEnum,
  Tooltip,
  Truncate,
  toast,
} from '@qovery/shared/ui'
import { dateUTCString, timeAgo } from '@qovery/shared/util-dates'
import { buildGitProviderUrl } from '@qovery/shared/util-git'
import { useCopyToClipboard } from '@qovery/shared/util-hooks'
import { containerRegistryKindToIcon, formatCronExpression, formatMetric, twMerge } from '@qovery/shared/util-js'
import useDeploymentStatus from '../hooks/use-deployment-status/use-deployment-status'
import useMasterCredentials from '../hooks/use-master-credentials/use-master-credentials'
import { useService } from '../hooks/use-service/use-service'
import { LastCommitAuthor } from '../last-commit-author/last-commit-author'
import { LastCommit } from '../last-commit/last-commit'
import { ServiceDetailsSkeleton } from './service-details-skeleton'

function GitRepository({
  service,
  gitRepository,
}: {
  service: Application | Job | Helm
  gitRepository: ApplicationGitRepository
}) {
  const { organizationId = '', projectId = '' } = useParams()

  return (
    <>
      {gitRepository.url && gitRepository.name && (
        <>
          <Dt>Repository:</Dt>
          <Dd>
            <a href={buildGitProviderUrl(gitRepository.url)} target="_blank" rel="noopener noreferrer">
              <Button color="neutral" variant="surface" size="xs" className="gap-1 text-nowrap">
                <Icon
                  name={
                    gitRepository.url.includes('//github')
                      ? IconEnum.GITHUB
                      : gitRepository.url.includes('//bitbucket')
                        ? IconEnum.BITBUCKET
                        : IconEnum.GITLAB
                  }
                  height={14}
                  width={14}
                />
                <Truncate text={gitRepository.name} truncateLimit={17} />
              </Button>
            </a>
          </Dd>
        </>
      )}
      {gitRepository.branch && gitRepository.url && (
        <>
          <Dt>Branch:</Dt>
          <Dd>
            <a
              href={buildGitProviderUrl(gitRepository.url, gitRepository.branch)}
              target="_blank"
              rel="noopener noreferrer"
            >
              <Button color="neutral" variant="surface" size="xs" className="gap-1">
                <Icon iconName="code-branch" iconStyle="regular" height={14} width={14} />
                <Truncate text={gitRepository.branch} truncateLimit={17} />
              </Button>
            </a>
          </Dd>
        </>
      )}
      <Dt>Target commit:</Dt>
      <Dd>
        <div className="inline-flex items-center gap-2">
          <LastCommitAuthor gitRepository={gitRepository} serviceId={service.id} serviceType={service.serviceType} />
          <LastCommit
            organizationId={organizationId}
            projectId={projectId}
            gitRepository={gitRepository}
            service={service}
          />
        </div>
      </Dd>
    </>
  )
}

export interface ServiceDetailsProps extends ComponentPropsWithoutRef<'div'> {
  environmentId: string
  serviceId: string
}

export function ServiceDetails({ className, environmentId, serviceId, ...props }: ServiceDetailsProps) {
  const { data: service } = useService({
    environmentId,
    serviceId,
  })
  const { data: deploymentStatus } = useDeploymentStatus({
    environmentId,
    serviceId,
  })

  const { data: masterCredentials } = useMasterCredentials({ serviceId, serviceType: service?.serviceType })
  const [, copyToClipboard] = useCopyToClipboard()

  if (!service) {
    return <ServiceDetailsSkeleton className={className} {...props} />
  }

  const containerImage = match(service)
    .with({ serviceType: ServiceTypeEnum.JOB, source: P.when(isJobContainerSource) }, ({ source }) => source.image)
    .with({ serviceType: ServiceTypeEnum.CONTAINER }, ({ image_name, tag, registry }) => ({
      image_name,
      tag,
      registry,
    }))
    .otherwise(() => undefined)

  const helmRepository = match(service)
    .with({ serviceType: 'HELM', source: P.when(isHelmRepositorySource) }, ({ source }) => source.repository)
    .otherwise(() => undefined)

  const databaseSource = match(service)
    .with({ serviceType: ServiceTypeEnum.DATABASE }, ({ accessibility, mode, type, version }) => ({
      accessibility,
      mode,
      type,
      version,
      masterCredentials,
    }))
    .otherwise(() => undefined)

  const ResourceUnit = ({ value, description }: { value?: string | number | null; description: string }) => {
    if (value === undefined || value === null) {
      return
    }
    return (
      <div className="grid-row-2 grid">
        <span className="text-base font-semibold text-neutral-400">{value}</span>
        <span className="text-xs">{description}</span>
      </div>
    )
  }

  const resources = match(service)
    .with(
      { serviceType: ServiceTypeEnum.CONTAINER },
      { serviceType: ServiceTypeEnum.APPLICATION },
      ({ cpu, memory, min_running_instances, max_running_instances }) => (
        <>
          <ResourceUnit
            value={
              Number.isNaN(min_running_instances) || Number.isNaN(max_running_instances)
                ? undefined
                : `${min_running_instances}/${max_running_instances}`
            }
            description="Instances (min/max)"
          />
          <ResourceUnit value={cpu && formatMetric({ current: cpu, unit: 'mCPU' })} description="vCPU (max)" />
          <ResourceUnit value={memory && formatMetric({ current: memory, unit: 'MiB' })} description="Memory (max)" />
        </>
      )
    )
    .with({ serviceType: ServiceTypeEnum.DATABASE }, ({ cpu, memory, storage, instance_type, mode }) => (
      <>
        {mode !== 'MANAGED' && (
          <>
            <ResourceUnit value={cpu && formatMetric({ current: cpu, unit: 'mCPU' })} description="vCPU (max)" />
            <ResourceUnit value={memory && formatMetric({ current: memory, unit: 'MiB' })} description="Memory (max)" />
          </>
        )}
        <ResourceUnit value={storage && formatMetric({ current: storage, unit: 'GiB' })} description="Storage (max)" />
        {mode !== 'CONTAINER' && <ResourceUnit value={instance_type} description="Instance type" />}
      </>
    ))
    .with({ serviceType: ServiceTypeEnum.JOB }, (job) => {
      const { cpu, memory, max_duration_seconds, max_nb_restart, port } = job
      return (
        <>
          {match(job)
            .with({ job_type: 'LIFECYCLE' }, ({ schedule }) => (
              <ResourceUnit
                value={
                  [schedule.on_start && 'Deploy', schedule.on_stop && 'Stop', schedule.on_delete && 'Delete']
                    .filter(Boolean)
                    .join(' - ') || undefined
                }
                description="Triggered on"
              />
            ))
            .with({ job_type: 'CRON' }, ({ schedule }) => (
              <ResourceUnit
                value={formatCronExpression(schedule.cronjob?.scheduled_at)}
                description={`Scheduling (${schedule.cronjob?.timezone})`}
              />
            ))
            .exhaustive()}
          <ResourceUnit value={max_nb_restart} description="Restart (max)" />
          <ResourceUnit value={max_duration_seconds && `${max_duration_seconds} s`} description="Duration (max)" />
          <ResourceUnit value={cpu && formatMetric({ current: cpu, unit: 'mCPU' })} description="vCPU (max)" />
          <ResourceUnit value={memory && formatMetric({ current: memory, unit: 'MiB' })} description="Memory (max)" />
          <ResourceUnit value={port} description="Port" />
        </>
      )
    })
    .otherwise(() => null)

  const valuesOverride = match(service)
    .with({ serviceType: 'HELM' }, (service) => {
      const {
        values_override: { file, set, set_json, set_string },
      } = service
      const overrideWithArguments = (
        <>
          <Dt>Override with arguments:</Dt>
          <Dl>{set?.length || set_json?.length || set_string?.length ? 'Yes' : 'No'}</Dl>
        </>
      )
      if (file?.git?.git_repository) {
        return (
          <Dl>
            <Dt>Type:</Dt>
            <Dd>Git repository</Dd>
            <GitRepository service={service} gitRepository={file.git.git_repository} />
            {overrideWithArguments}
          </Dl>
        )
      } else if (file?.raw) {
        return (
          <Dl>
            <Dt>Type:</Dt>
            <Dd>Raw YAML</Dd>
            {overrideWithArguments}
          </Dl>
        )
      } else {
        return <Dl>{overrideWithArguments}</Dl>
      }
    })
    .otherwise(() => null)

  const sectionClassName = 'text-neutral-350 gap-4 pl-8 pr-5'

  const handleCopyCredentials = (credentials: Credentials) => {
    const connectionURI = `${credentials?.login}:${credentials?.password}@${credentials?.host}:${credentials?.port}`
    copyToClipboard(connectionURI)
    toast(ToastEnum.SUCCESS, 'Credentials copied to clipboard')
  }

  return (
    <div className={twMerge('flex shrink-0 flex-col gap-5 py-8 text-sm', className)} {...props}>
      <Section className={sectionClassName}>
        <div className="flex flex-row justify-between">
          {/* XXX: Should be Heading, typography & design wanted */}
          <span className="font-medium text-neutral-400">About</span>
          <Link color="current" to={`..${APPLICATION_SETTINGS_URL + APPLICATION_SETTINGS_GENERAL_URL}`} relative="path">
            <Icon iconName="gear" iconStyle="regular" className="text-base text-neutral-350" />
          </Link>
        </div>
        <p>{service.description || 'No description provided yet'}</p>
        <Dl>
          {deploymentStatus?.last_deployment_date && (
            <>
              <Dt>Last deployment:</Dt>
              <Dd>
                <Tooltip content={dateUTCString(deploymentStatus.last_deployment_date)}>
                  <span>{timeAgo(new Date(deploymentStatus.last_deployment_date))}</span>
                </Tooltip>
              </Dd>
            </>
          )}
          {service.updated_at && (
            <>
              <Dt>Last edit:</Dt>
              <Dd>
                <Tooltip content={dateUTCString(service.updated_at)}>
                  <span>{timeAgo(new Date(service.updated_at))}</span>
                </Tooltip>
              </Dd>
            </>
          )}
          <Dt>Created:</Dt>
          <Dd>
            <Tooltip content={dateUTCString(service.created_at)}>
              <span>{timeAgo(new Date(service.created_at))}</span>
            </Tooltip>
          </Dd>
        </Dl>
      </Section>
      <hr />
      <Section className={sectionClassName}>
        <div className="flex flex-row justify-between">
          {/* XXX: Should be Heading, typography & design wanted */}
          <span className="font-medium text-neutral-400">Source</span>
          <Link color="current" to={`..${APPLICATION_SETTINGS_URL + APPLICATION_SETTINGS_GENERAL_URL}`} relative="path">
            <Icon iconName="gear" iconStyle="regular" className="text-base text-neutral-350" />
          </Link>
        </div>
        {match(service)
          .with(
            { serviceType: 'APPLICATION' },
            {
              serviceType: 'JOB',
              source: P.when(isJobGitSource),
            },
            {
              serviceType: 'HELM',
              source: P.when(isHelmGitSource),
            },
            (service) => {
              const gitRepository = match(service)
                .with({ serviceType: 'APPLICATION' }, ({ git_repository }) => git_repository)
                .with({ serviceType: 'JOB' }, ({ source }) => source.docker?.git_repository)
                .with({ serviceType: 'HELM' }, ({ source }) => source.git?.git_repository)
                .exhaustive()

              if (!gitRepository) {
                return null
              }

              return (
                <Dl>
                  <GitRepository service={service} gitRepository={gitRepository} />
                </Dl>
              )
            }
          )
          .otherwise(() => undefined)}
        {containerImage && (
          <Dl>
            {containerImage.registry && (
              <>
                <Dt>Registry:</Dt>
                <Dd>
                  <a
                    href={containerImage.registry.url}
                    target="_blank"
                    rel="noopener noreferrer"
                    className="max-w-full"
                  >
                    <Badge variant="surface" className="max-w-full items-center gap-1 whitespace-nowrap capitalize">
                      <Icon width={16} name={containerRegistryKindToIcon(containerImage.registry.kind)} />
                      <span className="truncate">
                        <Truncate text={containerImage.registry.name.toLowerCase()} truncateLimit={18} />
                      </span>
                    </Badge>
                  </a>
                </Dd>
              </>
            )}
            <Dt>Image name:</Dt>
            <Dd>{containerImage.image_name}</Dd>
            <Dt>Tag:</Dt>
            <Dd>
              <Truncate text={containerImage.tag} truncateLimit={18} />
            </Dd>
          </Dl>
        )}
        {helmRepository && (
          <Dl>
            {helmRepository.repository && (
              <>
                <Dt>Repository:</Dt>
                <Dd>
                  <a href={helmRepository.repository.url} target="_blank" rel="noopener noreferrer">
                    <Badge variant="surface" className="items-center gap-1 text-nowrap capitalize">
                      <Icon width={16} name={IconEnum.HELM_OFFICIAL} />
                      <Truncate text={helmRepository.repository.name ?? ''} truncateLimit={18} />
                    </Badge>
                  </a>
                </Dd>
              </>
            )}
            <Dt>Chart name:</Dt>
            <Dd>{helmRepository.chart_name}</Dd>
            <Dt>Target version:</Dt>
            <Dd>{helmRepository.chart_version}</Dd>
          </Dl>
        )}
        {databaseSource && (
          <Dl>
            <Dt>Type:</Dt>
            <Dd>
              <Badge variant="surface" className="items-center gap-1">
                <Icon name={databaseSource.type} className="max-h-[12px] max-w-[12px]" height={12} width={12} />
                {databaseSource.version}
              </Badge>
            </Dd>
            <Dt>Mode:</Dt>
            <Dd className="capitalize">{databaseSource.mode.toLowerCase()}</Dd>
            <Dt>Accessibility:</Dt>
            <Dd className="capitalize">{databaseSource.accessibility?.toLowerCase()}</Dd>
            {databaseSource.masterCredentials && (
              <>
                <Dt>Access:</Dt>
                <Dd>
                  <Button
                    color="neutral"
                    variant="surface"
                    size="xs"
                    onClick={() =>
                      /* XXX: TS is not able to infer that masterCredentials cannot be undef */
                      handleCopyCredentials(databaseSource.masterCredentials!)
                    }
                  >
                    <Icon iconName="key" iconStyle="regular" className="mr-2" />
                    Connection URI
                  </Button>
                </Dd>
              </>
            )}
          </Dl>
        )}
      </Section>
      {resources && (
        <>
          <hr />
          <Section className={sectionClassName}>
            <div className="flex flex-row justify-between">
              {/* XXX: Should be Heading, typography & design wanted */}
              <span className="flex items-center gap-1 font-medium text-neutral-400">
                Resources
                <Tooltip content="This is based on the configuration of your service">
                  <span className="flex items-center">
                    <Icon iconName="circle-info" iconStyle="regular" className="text-xs text-neutral-350" />
                  </span>
                </Tooltip>
              </span>
              <Link
                color="current"
                to={`..${APPLICATION_SETTINGS_URL + APPLICATION_SETTINGS_RESOURCES_URL}`}
                relative="path"
              >
                <Icon iconName="gear" iconStyle="regular" className="text-base text-neutral-350" />
              </Link>
            </div>
            <div className="grid grid-cols-2 gap-2">{resources}</div>
          </Section>
        </>
      )}
      {valuesOverride && (
        <>
          <hr />
          <Section className={sectionClassName}>
            <div className="flex flex-row justify-between">
              {/* XXX: Should be Heading, typography & design wanted */}
              <span className="flex items-center gap-1 font-medium text-neutral-400">Values override</span>
              <Link
                color="current"
                to={`..${APPLICATION_SETTINGS_URL + APPLICATION_SETTINGS_VALUES_OVERRIDE_FILE_URL}`}
                relative="path"
              >
                <Icon iconName="gear" iconStyle="regular" className="text-base text-neutral-350" />
              </Link>
            </div>
            {valuesOverride}
          </Section>
        </>
      )}
    </div>
  )
}

export default ServiceDetails
