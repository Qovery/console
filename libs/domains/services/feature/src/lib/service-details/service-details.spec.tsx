import { renderWithProviders } from '@qovery/shared/util-tests'
import { ServiceDetails } from './service-details'

jest.mock('../hooks/use-service/use-service', () => ({
  useService: ({
    serviceId,
  }: {
    environmentId: string
    serviceId: 'application-mock' | 'database-mock' | 'job-mock'
  }) => {
    const mocks = {
      'application-mock': {
        id: 'ebb84aa8-91c2-40fb-916d-3a158db354b7',
        created_at: '2023-04-12T08:48:51.801049Z',
        updated_at: '2023-09-28T06:48:09.079032Z',
        environment: {
          id: '28c47145-c8e7-4b9d-8d9e-c65c95b48425',
        },
        auto_preview: true,
        maximum_cpu: 3560,
        maximum_memory: 15411,
        name: 'console',
        description: 'React Application the Qovery Console',
        build_mode: 'DOCKER',
        dockerfile_path: 'Dockerfile',
        arguments: [],
        entrypoint: '',
        cpu: 50,
        memory: 50,
        min_running_instances: 1,
        max_running_instances: 2,
        storage: [],
        ports: [
          {
            internal_port: 80,
            external_port: 443,
            publicly_accessible: true,
            is_default: true,
            protocol: 'HTTP',
            name: 'p80',
            id: '6c9052e8-ebda-439d-bcb2-3b0298d4b51a',
          },
        ],
        healthchecks: {
          readiness_probe: null,
          liveness_probe: null,
        },
        git_repository: {
          has_access: true,
          deployed_commit_id: 'ddd1cee35762e9b4bb95633c22193393ca3bb384',
          deployed_commit_date: '2023-09-28T06:42:35.688665Z',
          deployed_commit_contributor: 'TAGS_NOT_IMPLEMENTED',
          deployed_commit_tag: 'TAGS_NOT_IMPLEMENTED',
          provider: 'GITHUB',
          owner: 'acarranoqovery',
          url: 'https://github.com/Qovery/console.git',
          name: 'Qovery/console',
          branch: 'staging',
          root_path: '/',
        },
        auto_deploy: true,
      },
      'database-mock': {
        id: 'ee3523e9-c81d-42ac-9d0c-f7bc09d5d28c',
        created_at: '2023-07-28T14:50:09.325974Z',
        updated_at: '2023-07-28T14:50:09.325976Z',
        environment: {
          id: '0cd5d05e-0839-48ff-be67-ca3f4fcf8250',
        },
        name: 'containered-posgresSQL-clone',
        description: '',
        type: 'POSTGRESQL',
        version: '15',
        mode: 'CONTAINER',
        disk_encrypted: false,
        accessibility: 'PRIVATE',
        host: 'zee3523e9-postgresql.zc531a994.rustrocks.cloud',
        port: 5432,
        cpu: 500,
        memory: 512,
        storage: 10,
        maximum_cpu: 4000,
        maximum_memory: 16384,
        instance_type: null,
      },
      'job-mock': {
        id: 'c070ebf8-5b82-4d94-8c4d-0c6b86d7c003',
        created_at: '2023-09-12T10:08:39.122972Z',
        updated_at: '2023-09-27T12:17:25.956823Z',
        environment: {
          id: '7aaa3a79-0afa-4d5e-b898-c2bf6f33a01a',
        },
        name: 'test_lifecycle',
        description: '',
        auto_preview: true,
        cpu: 50,
        memory: 51,
        max_nb_restart: 0,
        max_duration_seconds: 300,
        port: null,
        source: {
          docker: {
            git_repository: {
              has_access: true,
              deployed_commit_id: 'a82fbbb9ff34784d1b77bebede6c5ac909d3a865',
              deployed_commit_date: '2023-09-22T08:26:53.583504Z',
              deployed_commit_contributor: 'TAGS_NOT_IMPLEMENTED',
              deployed_commit_tag: 'TAGS_NOT_IMPLEMENTED',
              provider: 'GITHUB',
              owner: 'acarranoqovery',
              url: 'https://github.com/acarranoqovery/test_script.git',
              name: 'acarranoqovery/test_script',
              branch: 'main',
              root_path: '/',
            },
            dockerfile_path: 'Dockerfile',
          },
        },
        schedule: {
          on_start: {
            arguments: ['job.py'],
          },
        },
        maximum_cpu: 16000,
        maximum_memory: 16384,
        auto_deploy: false,
        healthchecks: {
          readiness_probe: null,
          liveness_probe: null,
        },
      },
    }
    return {
      data: mocks[serviceId],
      isLoading: false,
      error: null,
    }
  },
}))

jest.mock('../hooks/use-commits/use-commits', () => ({
  useCommits: () => ({
    data: [
      {
        created_at: '2023-09-28T06:42:32Z',
        git_commit_id: 'ddd1cee35762e9b4bb95633c22193393ca3bb384',
        message: 'refactor(ui): rename legacy button props (#863)',
        tag: 'staging',
        author_name: 'GitHub',
        author_avatar_url: 'https://avatars.githubusercontent.com/u/19864447?v=4',
        commit_page_url: 'https://github.com/Qovery/console/commit/ddd1cee35762e9b4bb95633c22193393ca3bb384',
      },
      {
        created_at: '2023-09-27T14:58:59Z',
        git_commit_id: '136b77f8a57026d1f4760edc84dc8999a5334f97',
        message: 'refactor(ui): rename legacy button types (#862)',
        tag: 'staging',
        author_name: 'GitHub',
        author_avatar_url: 'https://avatars.githubusercontent.com/u/19864447?v=4',
        commit_page_url: 'https://github.com/Qovery/console/commit/136b77f8a57026d1f4760edc84dc8999a5334f97',
      },
    ],
    isLoading: false,
    error: null,
  }),
}))

describe('ServiceDetails', () => {
  const now = new Date('2023-09-29:00:00Z')

  beforeAll(() => {
    jest.useFakeTimers()
    jest.setSystemTime(now)
  })

  it('should match database snapshot', () => {
    const { baseElement } = renderWithProviders(<ServiceDetails environmentId="1" serviceId="database-mock" />)
    expect(baseElement).toMatchSnapshot()
  })
  it('should match application snapshot', () => {
    const { baseElement } = renderWithProviders(<ServiceDetails environmentId="1" serviceId="application-mock" />)
    expect(baseElement).toMatchSnapshot()
  })
  it('should match job snapshot', () => {
    const { baseElement } = renderWithProviders(<ServiceDetails environmentId="1" serviceId="job-mock" />)
    expect(baseElement).toMatchSnapshot()
  })
})
