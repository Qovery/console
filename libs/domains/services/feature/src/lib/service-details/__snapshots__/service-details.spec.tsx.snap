// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceDetails should match application snapshot 1`] = `
<body>
  <div>
    <div
      class="flex shrink-0 flex-col gap-5 py-8 text-sm"
    >
      <section
        class="flex flex-col text-neutral-350 gap-4 pl-8 pr-5"
      >
        <div
          class="flex flex-row justify-between"
        >
          <span
            class="font-medium text-neutral-400"
          >
            About
          </span>
          <a
            class="cursor-pointer transition duration-100 font-medium inline-flex items-center gap-1 text-current hover:brightness-75 text-sm"
            href="/settings/general"
          >
            <i
              aria-hidden="true"
              class="fa-regular fa-gear text-base text-neutral-350"
            />
          </a>
        </div>
        <p>
          React Application the Qovery Console
        </p>
        <dl
          class="grid grid-cols-2 gap-2"
        >
          <dt
            class="text-neutral-350 dark:text-neutral-300"
          >
            Last edit:
          </dt>
          <dd
            class="font-medium text-neutral-400 dark:font-normal dark:text-white"
          >
            <span
              class=""
              data-state="closed"
            >
              17 hours
            </span>
          </dd>
          <dt
            class="text-neutral-350 dark:text-neutral-300"
          >
            Created:
          </dt>
          <dd
            class="font-medium text-neutral-400 dark:font-normal dark:text-white"
          >
            <span
              class=""
              data-state="closed"
            >
              6 months
            </span>
          </dd>
        </dl>
      </section>
      <hr />
      <section
        class="flex flex-col text-neutral-350 gap-4 pl-8 pr-5"
      >
        <div
          class="flex flex-row justify-between"
        >
          <span
            class="font-medium text-neutral-400"
          >
            Source
          </span>
          <a
            class="cursor-pointer transition duration-100 font-medium inline-flex items-center gap-1 text-current hover:brightness-75 text-sm"
            href="/settings/general"
          >
            <i
              aria-hidden="true"
              class="fa-regular fa-gear text-base text-neutral-350"
            />
          </a>
        </div>
      </section>
    </div>
  </div>
</body>
`;

exports[`ServiceDetails should match database snapshot 1`] = `
<body>
  <div>
    <div
      class="flex shrink-0 flex-col gap-5 py-8 text-sm"
    >
      <section
        class="flex flex-col text-neutral-350 gap-4 pl-8 pr-5"
      >
        <div
          class="flex flex-row justify-between"
        >
          <span
            class="font-medium text-neutral-400"
          >
            About
          </span>
          <a
            class="cursor-pointer transition duration-100 font-medium inline-flex items-center gap-1 text-current hover:brightness-75 text-sm"
            href="/settings/general"
          >
            <i
              aria-hidden="true"
              class="fa-regular fa-gear text-base text-neutral-350"
            />
          </a>
        </div>
        <p>
          No description provided yet
        </p>
        <dl
          class="grid grid-cols-2 gap-2"
        >
          <dt
            class="text-neutral-350 dark:text-neutral-300"
          >
            Last edit:
          </dt>
          <dd
            class="font-medium text-neutral-400 dark:font-normal dark:text-white"
          >
            <span
              class=""
              data-state="closed"
            >
              2 months
            </span>
          </dd>
          <dt
            class="text-neutral-350 dark:text-neutral-300"
          >
            Created:
          </dt>
          <dd
            class="font-medium text-neutral-400 dark:font-normal dark:text-white"
          >
            <span
              class=""
              data-state="closed"
            >
              2 months
            </span>
          </dd>
        </dl>
      </section>
      <hr />
      <section
        class="flex flex-col text-neutral-350 gap-4 pl-8 pr-5"
      >
        <div
          class="flex flex-row justify-between"
        >
          <span
            class="font-medium text-neutral-400"
          >
            Source
          </span>
          <a
            class="cursor-pointer transition duration-100 font-medium inline-flex items-center gap-1 text-current hover:brightness-75 text-sm"
            href="/settings/general"
          >
            <i
              aria-hidden="true"
              class="fa-regular fa-gear text-base text-neutral-350"
            />
          </a>
        </div>
      </section>
    </div>
  </div>
</body>
`;

exports[`ServiceDetails should match job snapshot 1`] = `
<body>
  <div>
    <div
      class="flex shrink-0 flex-col gap-5 py-8 text-sm"
    >
      <section
        class="flex flex-col text-neutral-350 gap-4 pl-8 pr-5"
      >
        <div
          class="flex flex-row justify-between"
        >
          <span
            class="font-medium text-neutral-400"
          >
            About
          </span>
          <a
            class="cursor-pointer transition duration-100 font-medium inline-flex items-center gap-1 text-current hover:brightness-75 text-sm"
            href="/settings/general"
          >
            <i
              aria-hidden="true"
              class="fa-regular fa-gear text-base text-neutral-350"
            />
          </a>
        </div>
        <p>
          No description provided yet
        </p>
        <dl
          class="grid grid-cols-2 gap-2"
        >
          <dt
            class="text-neutral-350 dark:text-neutral-300"
          >
            Last edit:
          </dt>
          <dd
            class="font-medium text-neutral-400 dark:font-normal dark:text-white"
          >
            <span
              class=""
              data-state="closed"
            >
              1 day
            </span>
          </dd>
          <dt
            class="text-neutral-350 dark:text-neutral-300"
          >
            Created:
          </dt>
          <dd
            class="font-medium text-neutral-400 dark:font-normal dark:text-white"
          >
            <span
              class=""
              data-state="closed"
            >
              17 days
            </span>
          </dd>
        </dl>
      </section>
      <hr />
      <section
        class="flex flex-col text-neutral-350 gap-4 pl-8 pr-5"
      >
        <div
          class="flex flex-row justify-between"
        >
          <span
            class="font-medium text-neutral-400"
          >
            Source
          </span>
          <a
            class="cursor-pointer transition duration-100 font-medium inline-flex items-center gap-1 text-current hover:brightness-75 text-sm"
            href="/settings/general"
          >
            <i
              aria-hidden="true"
              class="fa-regular fa-gear text-base text-neutral-350"
            />
          </a>
        </div>
      </section>
    </div>
  </div>
</body>
`;
